### PR TITLE
Document refactor

### PR DESCRIPTION
**Pull request by ArtSuslov**

### CODE REVIEW

File: src/base/util/upload-documents.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { Document } from '../document'
+import isObject from 'lodash/isObject'
+
+/**
+ * Uploads an array of documents to the storage.
+ *
+ * @param {Document[]} documents - array of {@link Document}
+ * instances to be uploaded
+ * @param {string} [accountId] - documents' owner account ID
```
Comment (max-nes) [inline src/base/util/upload-documents.js:9]:
```suggestion
```

File: src/base/document.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+ * @class
+ */
+export class Document {
+  /**
+   * @param {object} opts
+   * @param {File} [opts.file] - file representing a document
+   * @param {string} [opts.name] - file name
+   * @param {string} [opts.mimeType] - file MIME type
+   * @param {string} [opts.key] - storage service file key
+   * @param {string} [opts.type] - storage service document type
```
Comment (max-nes) [inline src/base/document.js:14]:
```suggestion
   * @param {string} [type] - storage service document type
```

File: src/base/document.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+  toJSON () {
+    return {
+      mime_type: this._mimeType || '',
+      name: this._name || '',
+      key: this._key || ''
+    }
+  }
+
+  static toEmptyJSON () {
```
Comment (quinn71) [inline src/base/document.js:77]: it's redundant

File: src/base/document.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
+import { DocumentsManager } from '../api2'
```
Comment (quinn71) [inline src/base/document.js:1]: `/base` is a folder ONLY for core-related stuff, it shouldn't be here

File: src/base/document.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  static toEmptyJSON () {
+    return {
+      mime_type: '',
+      name: '',
+      key: ''
+    }
+  }
+
+  async uploadSelf () {
+    if (this.isUploaded) return this
```
Comment (quinn71) [inline src/base/document.js:86]: Oh, it's also did policies conversion before, now it seems to be missing

Comment (quinn71) [reply src/base/document.js:86]: we can pass them to `documentsManager` when initializing it, and do the conversion inside it

File: src/base/document.js (hunk 3 above)
Comment (ArtSuslov) [reply src/base/document.js:77]: removed

File: src/base/document.js (hunk 4 above)
Comment (ArtSuslov) [reply src/base/document.js:1]: moved to src/utils (both document and upload-documents)

File: src/base/document.js (hunk 5 above)
Comment (ArtSuslov) [reply src/base/document.js:86]: need to discuss

**Pull request merged**